### PR TITLE
Implemented empty_string rule

### PR DESCRIPTION
**Pull request by iDevid**
Rule description:
Prefer checking `isEmpty` over comparing `string` to \"\".

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,9 @@
 
 #### Enhancements
 
+* Adds `empty_string` opt-in rule.
```
Comment (ornithocoder) [inline CHANGELOG.md]: HI @iDevid! Take a look [here](https://github.com/realm/SwiftLint/blob/master/CONTRIBUTING.md#tracking-changes) for the format used on the changelog file. For instance, it must contain two trailing spaces on the last line describing the change.

File: Tests/SwiftLintFrameworkTests/EmptyStringTests.swift [hunk 2]
```diff
@@ -0,0 +1,30 @@
+//
```
Comment (ornithocoder) [inline Tests/SwiftLintFrameworkTests/EmptyStringTests.swift]: You don't need to include this test file. The two cases listed here are the same from `triggeringExamples`. SwiftLint builds tests for snippets included on `nonTriggeringExamples` and `triggeringExamples` and automatically runs them to make sure the rule does what it says it does.

File: CHANGELOG.md [hunk 3]
```diff
@@ -22,6 +22,9 @@
 
 #### Enhancements
 
+* Adds `empty_string` opt-in rule.  
+[Davide Sibilio](https://github.com/idevid)
```
Comment (marcelofabri) [inline CHANGELOG.md]: please add two spaces before your name (check other entries)

File: Source/SwiftLintFramework/Rules/EmptyStringRule.swift [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        triggeringExamples: [
+            "myString ↓== \"\"",
+            "myString ↓!= \"\""
+        ]
+    )
+
+    public func validate(file: File) -> [StyleViolation] {
+        let pattern = "\\b\\s*(==|!=)\\s*\"\""
+        let excludingKinds = SyntaxKind.commentKinds
+        return file.match(pattern: pattern, excludingSyntaxKinds: excludingKinds).map {
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/EmptyStringRule.swift]: you should use the `match` variation that takes the exact syntax kinds you want to match:

```
return file.match(pattern: pattern, with: [.string])
```

File: Source/SwiftLintFramework/Rules/EmptyStringRule.swift [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+public struct EmptyStringRule: ConfigurationProviderRule, OptInRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "empty_string",
+        name: "Empty String",
+        description: "Prefer checking `isEmpty` over comparing `string` to \"\".",
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/EmptyStringRule.swift]: what do you think about changing this to `"Prefer checking `isEmpty` over comparing `string` to an empty string literal`?"

Comment (iDevid) [reply Source/SwiftLintFramework/Rules/EmptyStringRule.swift]: I think that is better

File: Source/SwiftLintFramework/Rules/EmptyStringRule.swift (hunk 4 above)
Comment (iDevid) [reply Source/SwiftLintFramework/Rules/EmptyStringRule.swift]: fixed! Thanks!

**Comment by iDevid** [comment]
There are some problems with osscheck?

**Comment by marcelofabri** [comment]
> Are there some problems with osscheck?

Apparently, there's an issue with SSL on Circle CI: the `cocoapods` job is failing with the same reason and it's not only this PR that is failing. I'll try to investigate tomorrow.

**Comment by marcelofabri** [comment]
Also related: https://github.com/blog/2507-weak-cryptographic-standards-removed

**Comment by marcelofabri** [comment]
I've merged #2068 which should fix the CI failure. Can you please rebase this PR?

File: CHANGELOG.md [hunk 6]
```diff
@@ -22,6 +22,9 @@
 
 #### Enhancements
 
+* Adds `empty_string` opt-in rule.
```
Comment (ornithocoder) [inline CHANGELOG.md]: Please add your entry to the **Master** section of the change log. Version 0.25.0's already out.

**Comment by marcelofabri** [comment]
@iDevid Thanks!

**Pull request merged**